int32 tower_index
---

